specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,484 @@
+perform at least one measurement. The `%Qubit` and `%Result` data types hence
+must be supported by all backends. Qubits and results can occur only as
+arguments in function calls and are represented as a pointer of type `%Qubit*`
+and `%Result*` respectively, where the pointer itself identifies the qubit or
+result value rather than a memory location where the value is stored: a 64-bit
+integer constant is cast to the appropriate pointer type. A more detailed
+elaboration on the purpose of this representation is given in the next
+subsection. The integer constant that is cast must be in the interval `[0,
+nrQubits)` for `%Qubit*` and `[0, nrResults)` for `%Result*`, where `nrQubits`
+and `nrResults` are the required number of qubits and results defined by the
+corresponding [entry point attributes](#attributes). Since backends may look at
+the values of the `required_qubits` and `required_results` attributes to
+determine whether a program can be executed, it is recommended to index qubits
+and results consecutively such that there are no unused values within these
+ranges.
+
+### Qubit and Result Usage
+
+The amount of available memory on a QPU both with regards to qubits and
+potentially also with regards to classical memory for storing measurement
+results before they are read out and transmitted to another classical processor
+is commonly still fairly limited. Any memory - quantum or classical - that is
+used during quantum execution is not usually managed dynamically. Instead,
+operations are scheduled and resources are bound as part of compilation. How
+early in the process this happens varies, and QIR permits to express programs in
+a form that either defers allocation and management of such resources to later
+stages, or to directly identify individual qubits and results by a constant
+integer value as outlined above. This permits to accurately reflect application
+intent for a variety of frontends.
+
+Ultimately, it is up to the executing backend what data structure is associated
+with a qubit or result value. This gives a backend the freedom to, e.g., process
+measurement results asynchronously, or attach additional device data to qubits.
+Qubit and result values are correspondingly represented as opaque pointers in
+the bitcode, and a QIR program must not dereference such pointers, independent
+on whether they are merely bitcasts of integer constants as they are in the Base
+Profile program above, or whether they are created dynamically, meaning the
+value is managed by the executing backend.
+
+To execute a given bitcode, the backend needs to know how to process qubit and
+result pointers used by a program. This is achieved by storing metadata in the
+form of [module flags](#module-flags-metadata) in the bitcode. QIR does not make
+a type distinction or for example uses a different address space for the two
+kinds of pointers. This ensures that libraries and optimization passes that map
```

Comment:
" QIR does not make a type distinction or for example uses a different address space for the two kinds of pointers" => Not sure about the grammatical construct here.

Reply:
rephrased